### DISCUSSION TITLE
import useStore from vuex will make it undefineded

### DISCUSSION THREAD

**Discussion by chisaato** [quasarframework/quasar#9373] [General - Components / Directives / etc] [answered]
**Describe the bug**

Folling the docs of next version  
I should import `useStore` from `vuex`  
but if i use the following code, it will cause the `myStore` be undefinded.

here is the code

```ts
import { computed } from "vue";
import { useStore } from "vuex";
import { useRouter } from "vue-router";

export default {
	data() {
		return {
			leftDrawerOpen: false,
			links1: [
				{ icon: "web", text: "项目管理" },
				{ icon: "person", text: "问题追踪" },
			],
		};
	},
	mounted() {
		const router = useRouter();
		if (this.tokenState === "") {
			// 说明还没有准备 Token,自然就是去登陆
			// console.log("还没有登陆");
			router.push({ path: "/login" });
		}
	},
	setup() {
		const myStore = useStore();
		console.log(myStore);
		const tokenState = computed({
			// 相当于重写了返回值,参考 Kotlin
			set: (value) => {
				myStore.commit("token/setBearerToken", value);
			},
			// get: () => myStore.getters["token/getBearerToken"],
			get: () => myStore.state.token.bearerToken,
		});
		return {
			tokenState,
		};
	},
};
```

if I import from `../store` will be ok  
and here is the doc

https://next.quasar.dev/quasar-cli/vuex-store#adding-a-vuex-module-

should I import useState from vuex or from `../state` ?  
is this an error of docs or just my fault?

**To Reproduce**

follow the docs to crate and use store.  
Ensure importing `useStore` from vuex

```ts
import { useStore } from "vuex";
```

then use the state in `setup()`

```ts
setup() {
		const myStore = useStore();
        // here will get undefinded
		console.log(myStore);
		const tokenState = computed({
			set: (value) => {
				myStore.commit("token/setBearerToken", value);
			},
            // here will fail
			get: () => myStore.state.token.bearerToken,
		});
		return {
			tokenState,
		};
	},
```

**Expected behavior**

**Screenshots**

![image](https://user-images.githubusercontent.com/17796830/119192749-32da1f00-bab3-11eb-8897-29f9d66774d7.png)

then the browser will say `Cannot read property 'state' of undefined` .  
if I tried to log what is `myStore` it will return undefined

![image](https://user-images.githubusercontent.com/17796830/119193538-58b3f380-bab4-11eb-88e5-4aac95f978ce.png)

the content of doc
**Platform (please complete the following information):**
Quasar Version: v2.0.0-beta.17
@quasar/app Version: v3.0.0-beta.25
Quasar mode:

- [x] SPA
- [ ] SSR
- [ ] PWA
- [ ] Electron
- [ ] Cordova
- [ ] Capacitor
- [ ] BEX

Tested on:

- [x] SPA
- [ ] SSR
- [ ] PWA
- [ ] Electron
- [ ] Cordova
- [ ] Capacitor
- [ ] BEX

OS: Fedora 33
Node: v14.17.0
NPM: 6.14.13
Yarn: 1.22.10
Browsers: Chrome latest
iOS: N/A
Android: N/A
Electron: N/A

**Additional context**
found alternative problem here  
https://github.com/quasarframework/quasar/issues/4768

**Comment by TobyMosque** [comment]
unable to reproduce:
https://codepen.io/tobias-mesquita/pen/vYxxLLX?editors=1010

**Comment by chisaato** [comment]
creating store from ` new Vuex.Store()` will be ok.
have you follow the docs and use it in project instead of codepen ?

**Comment by rstoenescu** [comment]
One thing that is out of place (at first glance) in the code that you pasted is that you're using a composition method inside of the `mounted()` hook (useRouter). A big "NO". In your `mounted()` you should use `this.$router`/`this.$route` instead (since the mounted hook is part of Options API). **Composition methods should be used exclusively inside of the `setup()` method**. Also, it's best to not mix Composition API with Options API.

**Reply by chisaato** [reply]
what's the equivalent replacement of mount in Composition api ?

**Comment by rstoenescu** [comment]
We can convert this into a GH Discussion if you want (post a reply to do so). So closing this ticket as it has nothing to do with an eventual bug in Quasar.

**Comment by chisaato** [comment]
sure， how to convert this issue to discussion？

**Reply by rstoenescu** [reply]
done.

**Comment by chisaato** [comment]
problem solved. cause `quasar new store`  put a storeKey and use it to create the store
![image](https://user-images.githubusercontent.com/17796830/119234417-63c45d80-bb60-11eb-8be8-35f0bfd5fa05.png)
then if i use `useStore()` without the storeKey it will give me undefinded

**Comment by chisaato** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/17796830/119234491-bf8ee680-bb60-11eb-9f73-7ee464e05981.png)

here will make undefinded 
to the solve is to create store without the storeKey in `src/store/index.ts`

```ts
export function useStore() {
  // return vuexUseStore(storeKey);
  return vuexUseStore();
}

```

or import the key from `../store` in vue template

![image](https://user-images.githubusercontent.com/17796830/119234569-3af09800-bb61-11eb-9fbf-68bec274a1ef.png)


but the storeKey is automatic generated, I think  we should point it out in the docs

**Reply by metalsadman** [reply]
why not just use the exposed `useStore` function from that file instead of the vuex one, so you don't have to pass the key everytime you use the useStore composable. ie. `import { useStore } from 'src/store/index'`. this is actually taken from vuex docs here https://next.vuex.vuejs.org/guide/typescript-support.html.

**Comment by tampler** [comment]
@metalsadman 

Had the same problem with Quasar 2.0.1 and TS. Fixed by following your advice and doing: `import { useStore } from 'src/store'`.

Thanks for the help!
